WPThemeReview/Sniffs/Templates/RequiredFunctionSniff.php

```diff
@@ -0,0 +1,231 @@
+ *
+ * @since 0.1.0
+ */
+class RequiredFunctionSniff implements Sniff {
+
+	/**
+	 * Sniff Settings
+	 *
+	 * @var array
+	 */
+	public $tagsConfig = array(
+		'body' => array(
+			'function'  => 'body_class',
+			'attribute' => 'class',
+		),
+		'html' => array(
+			'function'  => 'language_attributes',
+			'attribute' => 'lang',
+		),
+	);
+
+	/**
+	 * Supported Tokenizers
+	 *
+	 * Currently this sniff is only useful in PHP as the required
+	 * functions to call are done in PHP. In testing various
+	 * themes - some had inline comments including `<html>`, and
+	 * were tokenized as T_INLINE_HTML throwing some false positives.
+	 *
+	 * @var array
+	 */
+	public $supportedTokenizers = array( 'PHP' );
+
+	/**
+	 * Tag being searched.
+	 *
+	 * @var array
+	 */
+	protected $tag;
+
+	/**
+	 * Returns an array of tokens this test wants to listen for.
+	 *
+	 * @return array
+	 */
+	public function register() {
+		return Tokens::$textStringTokens;
+	}
+
+	/**
+	 * Processes this test, when one of its tokens is encountered.
+	 *
+	 * @param \PHP_CodeSniffer\Files\File $phpcsFile The PHP_CodeSniffer file where the
+	 *                                               token was found.
+	 * @param int                         $stackPtr  The position of the current token
+	 *                                               in the stack.
+	 *
+	 * @return void
+	 */
+	public function process( File $phpcsFile, $stackPtr ) {
+
+		$tokens   = $phpcsFile->getTokens();
+		$content  = $this->clean_str( $tokens[ $stackPtr ]['content'] );
+		$filename = $phpcsFile->getFileName();
+
+		// Set to false if it is the first time this sniff is run on a file.
+		if ( ! isset( $this->tag[ $filename ] ) ) {
+			$this->tag[ $filename ] = false;
+		}
+
+		// Skip on empty.
+		if ( '' === $content ) {
+			return;
+		}
+
+		// Set tag class property.
+		foreach ( $this->tagsConfig as $tag => $settings ) {
+
+			// HTML case should be insensitive.
+			if ( false !== stripos( $content, '<' . $tag ) ) {
+				$this->tag[ $filename ]        = $this->tagsConfig[ $tag ];
+				$this->tag[ $filename ]['tag'] = $tag;
+				break;
+			}
+		}
+
+		// Skip if not a tag.
+		if ( false === $this->tag[ $filename ] ) {
+			return;
+		}
+
+		// Set vars used for reference.
+		$tagName        = $this->tag[ $filename ]['tag'];
+		$tagFn          = $this->tag[ $filename ]['function'];
+		$tagAttr        = $this->tag[ $filename ]['attribute'];
+		$pascal         = str_replace( ' ', '', ucwords( str_replace( '_', ' ', $tagFn ) ) );
+		$nextPtr        = $stackPtr;
+		$foundFunction  = false;
+		$foundAttribute = false;
+		$foundEnd       = false;
+
+		do {
+			$nextPtrContent = $this->clean_str( $tokens[ $nextPtr ]['content'] );
+			$nextPtrCode    = $tokens[ $nextPtr ]['code'];
+
+			// Check for attribute not allowed.
+			if (
+				false === $foundAttribute &&
+				in_array( $nextPtrCode, Tokens::$textStringTokens, true ) &&
```

Comment:
You can replace the `in_array` check with

```php
isset( Tokens::$textStringTokens[ $nextPtrCode ] )
```

More performant, since the `Tokens::$textStringTokens` is a `key => value` array 🙂

Reply:
Ah great catch!